native/src/main/java/io/ballerina/stdlib/http/api/HttpCallableUnitCallback.java

```diff
@@ -73,17 +81,59 @@ public void notifySuccess(Object result) {
             invokeErrorInterceptors((BError) result, true);
             return;
         }
-        returnResponse(result);
+        returnResponse(result, checkAddingLinks(result));
     }
 
-    private void returnResponse(Object result) {
-        Object[] paramFeed = new Object[6];
+    private boolean checkAddingLinks(Object result) {
```

Comment:
Are we using the return type of this method anywhere ? If not we could remove it. If we use it shall we rename the method name to provide the complete description of the functionality ?

Reply:
Yes we are using this returned value to indicate whether add links succeed or not
How would you like to change its name? I thought `checkAddingLinks` was enough to describe the functionality

Reply:
I would separate the logic into to components. 
1. Check the possibility to add the link to internal mapping
2. Add the link to internal mapping

Reply:
Refactored the code in [c08fcec](https://github.com/ballerina-platform/module-ballerina-http/pull/1068/commits/c08fcecc7d972b5883ac1eddc4e946b95e8a9836)